src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1174,6 +1174,16 @@ internal static bool IsPrimitiveRecursiveStruct(this TypeSymbol t)
         /// </summary>
         internal static int ComputeHashCode(this NamedTypeSymbol type)
         {
+            Debug.Assert(type.Equals(type.OriginalDefinition, TypeCompareKind.CLRSignatureCompareOptions) == wasConstructedForAnnotations(type));
+
+            if (wasConstructedForAnnotations(type))
+            {
+                // A type that uses its own type parameters as type arguments was constructed only for the purpose of adding annotations
+                // In that case we'll want to compute a hash that ignores such annotations and matches the object-hash from the definition
```

Comment:
>object-hash [](start = 107, length = 11)

I think we should simply say "hash code fro the definition", whatever that hash code might be.